samples/custom_dispatch/cpu/mlp_plugin/CMakeLists.txt

```diff
@@ -108,4 +113,65 @@ iree_lit_test_suite(
     "hostonly"
 )
 
+# Append the includes used for this file to the pdll_compilation_commands
+file(APPEND ${CMAKE_BINARY_DIR}/pdll_compile_commands.yml
+  "--- !FileInfo:\n"
+  "  filepath: \"${CMAKE_CURRENT_SOURCE_DIR}/mlp_all_specs.pdll\"\n"
+  "  includes: \"${TORCH_MLIR_ROOT_DIR}/include;${CMAKE_CURRENT_SOURCE_DIR};${MLIR_INCLUDE_DIRS}\"\n"
+)
+
+iree_lit_test_suite(
+  NAME
+    mlp_pdll_linalg_example
+  SRCS
+    "mlp_linalg_pdll.mlir"
+  TOOLS
+    FileCheck
+    iree-compile
+    iree-run-module
+    mlir-pdll
```

Comment:
We don't distribute mlir-pdll with IREE. Can this use something we already distribute instead? I see a `--iree-preprocessing-pdl-spec-filename=` flag used in some other tests.

We could look into distributing mlir-pdll or some iree-pdll, but I'd need to see more information about the architecture and data flows involved here. The tablegen deps in `samples/custom_dispatch/cpu/mlp_plugin/mlp_all_specs.pdll` scare me... anything user facing shouldn't be touching compiler source files. Is this a compiler developer feature and not a user feature? Does the https://github.com/openxla/iree/tree/main/samples/custom_dispatch README cover this usage already? Could also document in the PR description.

Reply:
Yes, I agree. I talked with Mahesh and we decided on adding `--iree-preprocessing-pdll-spec-filename=`, which mirrors the flags for pdl and transform dialect. Is this in line with what you are thinking?

From my understanding, this is a compiler developer feature but @MaheshRavishankar might have a better answer. Also, the readme does not contain documentation which I can definitely add (I did add a readme to the `mlp_plugin` folder).

I have removed all the tablegen  include files since they aren't required when going from PDLL to PDL. However, the underlying LLVM PDLL utilities use the includes to preform error checking. So, if this is something people are going to spend time using It is probably worth having some way to add includes and tell iree where to find them.